app/src/Core/APII.php

```diff
@@ -2872,94 +2863,101 @@ public function retrieveArchive( $data )
 	 * @static
 	 * @return bool True if it is an archive.
 	 * @license https://www.gnu.org/licenses/agpl-3.0.txt
-	 * @copyright Copyright (c) 2015-2021, Maximilian Doerr, Internet Archive
-	 * @author Maximilian Doerr (Cyberpower678)
+	 * @copyright Copyright (c) 2015-2021, Maximilian Doerr, James Hare, Internet Archive
+	 * @author Maximilian Doerr (Cyberpower678), James Hare
 	 */
 	public static function isArchive( $url, &$data )
 	{
-		//A hacky check for HTML encoded pipes
-		$url         = str_replace( "&#124;", "|", $url );
-		$url         = preg_replace( '/#.*/', '', $url );
+		// A hacky check for HTML encoded pipes
+		$url = str_replace( "&#124;", "|", $url );
+		$url = preg_replace( '/#.*/', '', $url );
+
+		// Parsing the URL
 		$checkIfDead = new CheckIfDead();
-		$parts       = $checkIfDead->parseURL( $url );
+		$parts = $checkIfDead->parseURL( $url );
+
+		// First check: return false if no URL parts
 		if( empty( $parts['host'] ) ) return false;
-		if( strpos( $parts['host'], "europarchive.org" ) !== false ||
-		    strpos( $parts['host'], "internetmemory.org" ) !== false ) {
-			$resolvedData = self::resolveEuropaURL( $url );
-		} elseif( strpos( $parts['host'], "webarchive.org.uk" ) !== false ) {
-			$resolvedData = self::resolveUKWebArchiveURL( $url );
-		} elseif( strpos( $parts['host'], "archive.org" ) !== false ||
-		          strpos( $parts['host'], "waybackmachine.org" ) !== false
-		) {
-			$resolvedData = self::resolveWaybackURL( $url );
+
+		// Second check: resolve varying hosts through varying UrlResolvers.
+		$archiveIsVariants = [
+				"archive.is",
+				"archive.today",
+				"archive.fo",
+				"archive.li",
+				"archive.vn",
+				"archive.md",
+				"archive.ph"];
+
+		if ( self::hostMatch( $parts['host'], [ "europarchive.org", "internetmemory.org" ] ) ) {
+			$resolvedData = UrlResolver::resolveEuropa( $url );
+		} elseif ( self::hostMatch( $parts['host'], [ "webarchive.org.uk" ] ) ) {
+			$resolvedData = UrlResolver::resolveUKWebArchive( $url );
+		} elseif ( self::hostMatch( $parts['host'], [ "archive.org", "waybackmachine.org" ] ) ) {
```

Comment:
It should be web.archive.org.  Anything outside of it is not the Wayback Machine, and may be misinterpreted.  Try to be as specific as possible when it comes to domains.

Reply:
I note the old code didn't have it either, which is actually a bug. :p